utils/validate_pr_quickstarts.js

```diff
@@ -0,0 +1,359 @@
+ * @param {String} filePath - Full file path of a file in a quickstart.
+ * @param {String} targetChild - Node in file path that should be preceded by a quickstart directory.
+ * @return {String} Node in file path of the quickstart.
+ */
+const getQuickstartNode = (filePath, targetChild) => {
+  const splitFilePath = filePath.split('/');
+  return splitFilePath[
+    splitFilePath.findIndex((path) => path === targetChild) - 1
+  ];
+};
+
+/**
+ * Identifies where in a given file path to look for a quickstart directory.
+ * @param {String} filePath - Full file path of a file in a quickstart.
+ * @return {Function|undefined} Called function with arguments to determine the quickstart of a given file path.
+ */
+const getQuickstartFromFilename = (filePath) => {
+  if (!filePath.includes('quickstarts/')) {
+    return undefined;
+  }
+
+  if (filePath.includes('/alerts/')) {
+    return getQuickstartNode(filePath, 'alerts');
+  }
+
+  if (filePath.includes('/dashboards/')) {
+    return getQuickstartNode(filePath, 'dashboards');
+  }
+
+  if (filePath.includes('/images/')) {
+    return getQuickstartNode(filePath, 'images');
+  }
+
+  const targetChildNode = filePath.split('/').pop();
+
+  return getQuickstartNode(filePath, targetChildNode);
+};
+
+/**
+ * Looks up corresponding quickstart config files for quickstarts known to have changes in a PR.
+ * @param {Set} quickstartDirectories - Set of unique quickstart directories.
+ * @return {Array} Collection of config file paths.
+ */
+const getQuickstartConfigPaths = (quickstartDirectories) => {
+  const allQuickstartConfigPaths = findMainQuickstartConfigFiles();
+
+  return [...quickstartDirectories].reduce((acc, quickstartDirectory) => {
+    const match = allQuickstartConfigPaths.find((path) =>
+      path.split('/').includes(quickstartDirectory)
+    );
+    if (match) {
+      acc.push(match);
+    }
+    return acc;
+  }, []);
+};
+
+/**
+ * Builds input argument for submitQuickstart GraphQL mutation.
+ * @param {Object} quickstartConfig - An object containing the path and contents of a quickstart config file.
+ * @return {Object} An object that represents a quickstart in the context of a GraphQL mutation.
+ */
+const buildMutationVariables = (quickstartConfig) => {
+  const {
+    authors,
+    categoryTerms,
+    description,
+    title,
+    documentation,
+    logo,
+    keywords,
+    summary,
+    installPlans,
+    id,
+  } = quickstartConfig.contents[0] || {};
+  const alertConfigPaths = getQuickstartAlertsConfigs(quickstartConfig.path);
+  const dashboardConfigPaths = getQuickstartDashboardConfigs(
+    quickstartConfig.path
+  );
+
+  return {
+    id: id,
+    quickstartMetadata: {
+      alertConditions:
+        alertConfigPaths.length > 0
+          ? adaptQuickstartAlertsInput(alertConfigPaths)
+          : undefined,
+      authors: authors && authors.map((author) => ({ name: author })),
+      categoryTerms: categoryTerms || keywords,
```

Comment:
We came across a bit of unknown here. Would there ever be a key of `categoryTerms` from an open source `config.yml`?

Looks like the README points contributors exclusively to `keywords` https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#keywords

I think there might just be a bit of overlap in the catalog service ([internal repo ref](https://source.datanerd.us/nr1-dev-experience/catalog-service-elixir/blob/5400c85f2dc8042902e37e27ab0fde4f36280d89/apps/api/lib/api/resolvers/quickstart_metadata.ex#L46-L57)), but if that's not relevant for this repo we might as well remove this

Reply:
While we don't necessarily need to support that in the config yet, I'd opt for parsing them out of the keywords before submitting to the mutation. The mutation does not associate quickstarts with categories by looking at keywords, so we need to ensure they get submitted as such.

We essentially just need to copy over the `associatedKeywords` from [this list](https://github.com/newrelic/developer-website/blob/develop/src/data/instant-observability-categories.json) and pull out any keyword that matches that list into `categoryTerms` before submitting the mutation.

Reply:
Thanks @jerelmiller. 

Could I confirm I'm following the suggestion right before implementing this:

1. User submits a config with `keywords`
2. Workflow looks at those keywords from the user's config and match them to the category `value` via the `associatedKeywords` set
3. Workflow sends only the set of any category `value` strings in the `categoryTerms` key of the mutation variables this function is building up

Does that sound right?

Reply:
Thats correct!

Reply:
Cool, this is where I landed: 0a9e41ab6724b5cb88f5a09b185b424f30bf50aa

Ended up removing `categoryTerms` as a value taken from the user-supplied config since we're not expecting them to supply those categories and building the array of categories based on keywords.